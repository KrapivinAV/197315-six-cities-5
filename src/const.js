export const OfferListStyle = {
  MAIN_SCREEN: `cities__places-list places__list tabs__content`,
  PROPERTIES_SCREEN: `near-places__list places__list`,
  FAVORITES_SCREEN: `favorites__places`
};

export const OfferCardStyleSet = {
  MAIN_SCREEN: {
    ARTICLE: `cities__place-card`,
    IMAGE_WRAPPER: `cities__image-wrapper`,
    INFO: ``
  },
  PROPERTIES_SCREEN: {
    ARTICLE: `near-places__card`,
    IMAGE_WRAPPER: `near-places__image-wrapper`,
    INFO: ``
  },
  FAVORITES_SCREEN: {
    ARTICLE: `favorites__card`,
    IMAGE_WRAPPER: `favorites__image-wrapper`,
    INFO: `favorites__card-info`
  }
};

export const offerTypes = {
  apartment: `Apartment`,
  room: `Private Room`,
  house: `House`,
  hotel: `Hotel`
};

export const cities = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const AppRoute = {
  MAIN: `/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`,
  PROPERTIES: `/offer/:id`
};

export const NEAR_OFFERS_MAX_QUANTITY = 3;
export const REVIEWS_MAX_QUANTITY = 10;
