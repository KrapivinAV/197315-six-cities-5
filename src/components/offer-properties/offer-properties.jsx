import React from "react";
import PropTypes from "prop-types";
import PropTypesSet from "../../prop-types-set";
import CommentForm from "../comment-form/comment-form";
import ReviewsList from "../reviews-list/reviews-list";
import Map from "../map/map";
import PropertiesScreenOfferList from "../properties-screen-offer-list/properties-screen-offer-list";
import withCommentFormState from "../../hocs/with-comment-form-state/with-comment-form-state";
import {offerTypes, AuthorizationStatus} from "../../const";
import {connect} from "react-redux";
import {getOffer, getCurrentNearOffers, getCurrentReviews, getAuthorizationStatus} from "../../store/selectors/selectors";

const CommentFormWrapped = withCommentFormState(CommentForm);

const OfferProperties = ({offer, nearOffers, reviews, authorizationStatus}) => {
  const {title, isPremium, isFavorite, type, rating, price, images, bedrooms, maxAdults, goods, host, description} = offer;

  console.log(authorizationStatus);

  const naturalRating = `${Math.round(rating) * 20}%`;

  const premiumType = isPremium ?
    (
      <div className="property__mark">
        <span>Premium</span>
      </div>
    ) :
    null;

  const favoriteButtonStyle = isFavorite ?
    `property__bookmark-button property__bookmark-button--active button` :
    `property__bookmark-button button`;

  return (
    <main className="page__main page__main--property">
      <section className="property">
        <div className="property__gallery-container container">
          <div className="property__gallery">

            {images.map((image) => (
              <div key={image} className="property__image-wrapper">
                <img className="property__image" src={image} alt="Photo studio" />
              </div>
            ))}

          </div>
        </div>
        <div className="property__container container">
          <div className="property__wrapper">

            {premiumType}

            <div className="property__name-wrapper">
              <h1 className="property__name">
                {title}
              </h1>
              <button className={favoriteButtonStyle} type="button">
                <svg className="property__bookmark-icon" width="31" height="33">
                  <use xlinkHref="#icon-bookmark"></use>
                </svg>
                <span className="visually-hidden">To bookmarks</span>
              </button>
            </div>
            <div className="property__rating rating">
              <div className="property__stars rating__stars">
                <span style={{width: naturalRating}}></span>
                <span className="visually-hidden">Rating</span>
              </div>
              <span className="property__rating-value rating__value">{rating.toFixed(1)}</span>
            </div>
            <ul className="property__features">
              <li className="property__feature property__feature--entire">
                {offerTypes[type]}
              </li>
              <li className="property__feature property__feature--bedrooms">
                {bedrooms} Bedrooms
              </li>
              <li className="property__feature property__feature--adults">
                  Max {maxAdults} adults
              </li>
            </ul>
            <div className="property__price">
              <b className="property__price-value">&euro;{price}</b>
              <span className="property__price-text">&nbsp;night</span>
            </div>
            <div className="property__inside">
              <h2 className="property__inside-title">What&apos;s inside</h2>
              <ul className="property__inside-list">

                {goods.map((item) => (
                  <li key={item} className="property__inside-item">
                    {item}
                  </li>
                ))}

              </ul>
            </div>
            <div className="property__host">
              <h2 className="property__host-title">Meet the host</h2>
              <div className="property__host-user user">
                <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                  <img className="property__avatar user__avatar" src={host.avatarUrl} width="74" height="74" alt="Host avatar" />
                </div>
                <span className="property__user-name">
                  {host.name}
                </span>
              </div>
              <div className="property__description">
                <p className="property__text">
                  {description}
                </p>
              </div>
            </div>
            <section className="property__reviews reviews">
              <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{reviews.length}</span></h2>

              <ReviewsList offerReviews={reviews}/>

              {authorizationStatus === AuthorizationStatus.AUTH ? <CommentFormWrapped /> : null}

            </section>
          </div>
        </div>
        <section className="property__map map">

          <Map offers={[offer, ...nearOffers]}/>

        </section>
      </section>
      <div className="container">
        <section className="near-places places">
          <h2 className="near-places__title">Other places in the neighbourhood</h2>

          <PropertiesScreenOfferList offers={nearOffers} />

        </section>
      </div>
    </main>
  );
};

OfferProperties.propTypes = {
  authorizationStatus: PropTypes.string.isRequired,
  offer: PropTypesSet.offer.isRequired,
  nearOffers: PropTypes.arrayOf(PropTypesSet.offer).isRequired,
  reviews: PropTypes.arrayOf(PropTypesSet.review).isRequired,
};

const mapStateToProps = (state) => ({
  authorizationStatus: getAuthorizationStatus(state),
  offer: getOffer(state),
  nearOffers: getCurrentNearOffers(state),
  reviews: getCurrentReviews(state)
});

export {OfferProperties};
export default connect(mapStateToProps)(OfferProperties);
