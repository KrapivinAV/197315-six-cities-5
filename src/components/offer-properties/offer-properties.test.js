import React from "react";
import renderer from "react-test-renderer";
import {OfferProperties} from "./offer-properties";
import {Provider} from "react-redux";
import {Route, BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";
import {AuthorizationStatus} from "../../const";
import {OFFER_FIRST_CITY_IS_FAVORITE_IS_PREMIUM_HOST_IS_PRO, OFFER_FIRST_CITY_NOT_FAVORITE_NOT_PREMIUM_HOST_NOT_PRO, REVIEW_FIRST, REVIEW_SECOND} from "../../test-mocks";

const noop = () => {};

jest.mock(`../map/map`, () => `Map`);

describe(`Render OfferProperties`, () => {
  it(`With all flags UP and Auth`, () => {

    const mockStore = configureStore([]);
    const store = mockStore({
      DATA: {
        offer: OFFER_FIRST_CITY_IS_FAVORITE_IS_PREMIUM_HOST_IS_PRO,
        nearOffers: [
          OFFER_FIRST_CITY_NOT_FAVORITE_NOT_PREMIUM_HOST_NOT_PRO
        ],
        reviews: [
          REVIEW_FIRST,
          REVIEW_SECOND
        ],
      },
      STATE: {
        currentOfferCardId: `1`,
        currentCity: `Paris`
      },
      USER: {
        authorizationStatus: `AUTH`,
        userData: {
          email: `user@mail.ru`,
        }
      }
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Route>
                <OfferProperties
                  offer={OFFER_FIRST_CITY_IS_FAVORITE_IS_PREMIUM_HOST_IS_PRO}
                  nearOffers={[OFFER_FIRST_CITY_NOT_FAVORITE_NOT_PREMIUM_HOST_NOT_PRO]}
                  reviews={[REVIEW_FIRST, REVIEW_SECOND]}
                  authorizationStatus={AuthorizationStatus.AUTH}
                  changeOfferFavoriteStatusAction={noop}
                  redirectToRouteAction={noop}
                  addCommentAction={noop}
                />
              </Route>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`With all flags DOWN without Auth`, () => {

    const mockStore = configureStore([]);
    const store = mockStore({
      DATA: {
        offer: OFFER_FIRST_CITY_NOT_FAVORITE_NOT_PREMIUM_HOST_NOT_PRO,
        nearOffers: [
          OFFER_FIRST_CITY_IS_FAVORITE_IS_PREMIUM_HOST_IS_PRO
        ],
        reviews: [
          REVIEW_SECOND,
          REVIEW_FIRST
        ],
      },
      STATE: {
        currentOfferCardId: `1`,
        currentCity: `Paris`
      },
      USER: {
        authorizationStatus: `NO_AUTH`,
        userData: {
          email: `user@mail.ru`,
        }
      }
    });

    const tree = renderer
      .create(
          <Provider store={store}>
            <BrowserRouter>
              <Route>
                <OfferProperties
                  offer={OFFER_FIRST_CITY_NOT_FAVORITE_NOT_PREMIUM_HOST_NOT_PRO}
                  nearOffers={[OFFER_FIRST_CITY_IS_FAVORITE_IS_PREMIUM_HOST_IS_PRO]}
                  reviews={[REVIEW_FIRST, REVIEW_SECOND]}
                  authorizationStatus={AuthorizationStatus.NO_AUTH}
                  changeOfferFavoriteStatusAction={noop}
                  redirectToRouteAction={noop}
                  addCommentAction={noop}
                />
              </Route>
            </BrowserRouter>
          </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
