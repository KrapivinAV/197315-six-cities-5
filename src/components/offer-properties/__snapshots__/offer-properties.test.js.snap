// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render OfferProperties With all flags DOWN without Auth 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive
          </h1>
          <button
            className="property__bookmark-button button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.0
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            5
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            8
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            226
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Peaceful studio in the most wanted area in town. Quiet house Near of everything.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  ak
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  99999999999999999999999999999999999999999999999999999
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  November 2020
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  ak
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  99999999999999999999999999999999999999999999999999999
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  October 2020
                </time>
              </div>
            </li>
          </ul>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <Map
        offers={
          Array [
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything.",
              "goods": Array [
                "Dishwasher",
                "Fridge",
                "Towels",
              ],
              "host": Object {
                "avatarUrl": "/img/avatar-angelina.jpg",
                "id": 25,
                "isPro": false,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 48.87761,
                "longitude": 2.333499,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "price": 226,
              "rating": 3,
              "title": "The house among olive",
              "type": "apartment",
            },
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything.",
              "goods": Array [
                "Dishwasher",
                "Fridge",
                "Towels",
              ],
              "host": Object {
                "avatarUrl": "/img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 48.87761,
                "longitude": 2.333499,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "price": 226,
              "rating": 3,
              "title": "The house among olive",
              "type": "apartment",
            },
          ]
        }
        screenMarker="properties"
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="near-places__card place-card"
          id={1}
          onMouseOver={[Function]}
        >
          <div
            className="near-places__image-wrapper place-card__image-wrapper"
          >
            <a>
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className=" place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  226
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button place-card__bookmark-button--active button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                The house among olive
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;

exports[`Render OfferProperties With all flags UP and Auth 1`] = `
<main
  className="page__main page__main--property"
>
  <section
    className="property"
  >
    <div
      className="property__gallery-container container"
    >
      <div
        className="property__gallery"
      >
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg"
          />
        </div>
        <div
          className="property__image-wrapper"
        >
          <img
            alt="Photo studio"
            className="property__image"
            src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg"
          />
        </div>
      </div>
    </div>
    <div
      className="property__container container"
    >
      <div
        className="property__wrapper"
      >
        <div
          className="property__mark"
        >
          <span>
            Premium
          </span>
        </div>
        <div
          className="property__name-wrapper"
        >
          <h1
            className="property__name"
          >
            The house among olive
          </h1>
          <button
            className="property__bookmark-button property__bookmark-button--active button"
            onClick={[Function]}
            type="button"
          >
            <svg
              className="property__bookmark-icon"
              height="33"
              width="31"
            >
              <use
                xlinkHref="#icon-bookmark"
              />
            </svg>
            <span
              className="visually-hidden"
            >
              To bookmarks
            </span>
          </button>
        </div>
        <div
          className="property__rating rating"
        >
          <div
            className="property__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "60%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
          <span
            className="property__rating-value rating__value"
          >
            3.0
          </span>
        </div>
        <ul
          className="property__features"
        >
          <li
            className="property__feature property__feature--entire"
          >
            Apartment
          </li>
          <li
            className="property__feature property__feature--bedrooms"
          >
            5
             Bedrooms
          </li>
          <li
            className="property__feature property__feature--adults"
          >
            Max 
            8
             adults
          </li>
        </ul>
        <div
          className="property__price"
        >
          <b
            className="property__price-value"
          >
            €
            226
          </b>
          <span
            className="property__price-text"
          >
             night
          </span>
        </div>
        <div
          className="property__inside"
        >
          <h2
            className="property__inside-title"
          >
            What's inside
          </h2>
          <ul
            className="property__inside-list"
          >
            <li
              className="property__inside-item"
            >
              Dishwasher
            </li>
            <li
              className="property__inside-item"
            >
              Fridge
            </li>
            <li
              className="property__inside-item"
            >
              Towels
            </li>
          </ul>
        </div>
        <div
          className="property__host"
        >
          <h2
            className="property__host-title"
          >
            Meet the host
          </h2>
          <div
            className="property__host-user user"
          >
            <div
              className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
            >
              <img
                alt="Host avatar"
                className="property__avatar user__avatar"
                height="74"
                src="/img/avatar-angelina.jpg"
                width="74"
              />
            </div>
            <span
              className="property__user-name"
            >
              Angelina
            </span>
          </div>
          <div
            className="property__description"
          >
            <p
              className="property__text"
            >
              Peaceful studio in the most wanted area in town. Quiet house Near of everything.
            </p>
          </div>
        </div>
        <section
          className="property__reviews reviews"
        >
          <h2
            className="reviews__title"
          >
            Reviews · 
            <span
              className="reviews__amount"
            >
              2
            </span>
          </h2>
          <ul
            className="reviews__list"
          >
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  ak
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "60%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  99999999999999999999999999999999999999999999999999999
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  November 2020
                </time>
              </div>
            </li>
            <li
              className="reviews__item"
            >
              <div
                className="reviews__user user"
              >
                <div
                  className="reviews__avatar-wrapper user__avatar-wrapper"
                >
                  <img
                    alt="Reviews avatar"
                    className="reviews__avatar user__avatar"
                    height="54"
                    src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/1.jpg"
                    width="54"
                  />
                </div>
                <span
                  className="reviews__user-name"
                >
                  ak
                </span>
              </div>
              <div
                className="reviews__info"
              >
                <div
                  className="reviews__rating rating"
                >
                  <div
                    className="reviews__stars rating__stars"
                  >
                    <span
                      style={
                        Object {
                          "width": "100%",
                        }
                      }
                    />
                    <span
                      className="visually-hidden"
                    >
                      Rating
                    </span>
                  </div>
                </div>
                <p
                  className="reviews__text"
                >
                  99999999999999999999999999999999999999999999999999999
                </p>
                <time
                  className="reviews__time"
                  dateTime="2019-04-24"
                >
                  October 2020
                </time>
              </div>
            </li>
          </ul>
          <form
            action="#"
            className="reviews__form form"
            method="post"
            onSubmit={[Function]}
          >
            <label
              className="reviews__label form__label"
              htmlFor="review"
            >
              Your review
            </label>
            <div
              className="reviews__rating-form form__rating"
            >
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                id="5-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="5-stars"
                title="perfect"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                id="4-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="4-stars"
                title="good"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                id="3-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="3-stars"
                title="not bad"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                id="2-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="2-stars"
                title="badly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
              <input
                checked={false}
                className="form__rating-input visually-hidden"
                id="1-stars"
                name="rating"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="reviews__rating-label form__rating-label"
                htmlFor="1-stars"
                title="terribly"
              >
                <svg
                  className="form__star-image"
                  height="33"
                  width="37"
                >
                  <use
                    xlinkHref="#icon-star"
                  />
                </svg>
              </label>
            </div>
            <textarea
              className="reviews__textarea form__textarea"
              id="review"
              name="review"
              onChange={[Function]}
              placeholder="Tell how was your stay, what you like and what can be improved"
              value=""
            />
            <div
              className="reviews__button-wrapper"
            >
              <p
                className="reviews__help"
              >
                To submit review please make sure to set 
                <span
                  className="reviews__star"
                >
                  rating
                </span>
                and describe your stay with at least 
                <b
                  className="reviews__text-amount"
                >
                  50 characters
                </b>
                .
              </p>
              <button
                className="reviews__submit form__submit button"
                disabled="disabled"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </section>
      </div>
    </div>
    <section
      className="property__map map"
    >
      <Map
        offers={
          Array [
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything.",
              "goods": Array [
                "Dishwasher",
                "Fridge",
                "Towels",
              ],
              "host": Object {
                "avatarUrl": "/img/avatar-angelina.jpg",
                "id": 25,
                "isPro": true,
                "name": "Angelina",
              },
              "id": 1,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": true,
              "isPremium": true,
              "location": Object {
                "latitude": 48.87761,
                "longitude": 2.333499,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "price": 226,
              "rating": 3,
              "title": "The house among olive",
              "type": "apartment",
            },
            Object {
              "bedrooms": 5,
              "city": Object {
                "location": Object {
                  "latitude": 48.85661,
                  "longitude": 2.351499,
                  "zoom": 13,
                },
                "name": "Paris",
              },
              "description": "Peaceful studio in the most wanted area in town. Quiet house Near of everything.",
              "goods": Array [
                "Dishwasher",
                "Fridge",
                "Towels",
              ],
              "host": Object {
                "avatarUrl": "/img/avatar-angelina.jpg",
                "id": 25,
                "isPro": false,
                "name": "Angelina",
              },
              "id": 2,
              "images": Array [
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/19.jpg",
                "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/3.jpg",
              ],
              "isFavorite": false,
              "isPremium": false,
              "location": Object {
                "latitude": 48.87761,
                "longitude": 2.333499,
                "zoom": 16,
              },
              "maxAdults": 8,
              "previewImage": "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg",
              "price": 226,
              "rating": 3,
              "title": "The house among olive",
              "type": "apartment",
            },
          ]
        }
        screenMarker="properties"
      />
    </section>
  </section>
  <div
    className="container"
  >
    <section
      className="near-places places"
    >
      <h2
        className="near-places__title"
      >
        Other places in the neighbourhood
      </h2>
      <div
        className="near-places__list places__list"
      >
        <article
          className="near-places__card place-card"
          id={2}
          onMouseOver={[Function]}
        >
          <div
            className="near-places__image-wrapper place-card__image-wrapper"
          >
            <a>
              <img
                alt="Place image"
                className="place-card__image"
                height="200"
                src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/13.jpg"
                width="260"
              />
            </a>
          </div>
          <div
            className=" place-card__info"
          >
            <div
              className="place-card__price-wrapper"
            >
              <div
                className="place-card__price"
              >
                <b
                  className="place-card__price-value"
                >
                  €
                  226
                </b>
                <span
                  className="place-card__price-text"
                >
                  / night
                </span>
              </div>
              <button
                className="place-card__bookmark-button button"
                onClick={[Function]}
                type="button"
              >
                <svg
                  className="place-card__bookmark-icon"
                  height="19"
                  width="18"
                >
                  <use
                    xlinkHref="#icon-bookmark"
                  />
                </svg>
                <span
                  className="visually-hidden"
                >
                  In bookmarks
                </span>
              </button>
            </div>
            <div
              className="place-card__rating rating"
            >
              <div
                className="place-card__stars rating__stars"
              >
                <span
                  style={
                    Object {
                      "width": "60%",
                    }
                  }
                />
                <span
                  className="visually-hidden"
                >
                  Rating
                </span>
              </div>
            </div>
            <h2
              className="place-card__name"
            >
              <a
                href="/offer/2"
                onClick={[Function]}
              >
                The house among olive
              </a>
            </h2>
            <p
              className="place-card__type"
            >
              Apartment
            </p>
          </div>
        </article>
      </div>
    </section>
  </div>
</main>
`;
